src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/NodeFactory.cs

```diff
@@ -389,11 +389,11 @@ public BlobNode ReadOnlyDataBlob(string name, byte[] blobData, int alignment)
             return _readOnlyDataBlobs.GetOrAdd(new Tuple<string, byte[], int>(name, blobData, alignment));
         }
 
-        private NodeCache<EETypeOptionalFieldsBuilder, EETypeOptionalFieldsNode> _typeOptionalFields;
+        private NodeCache<EETypeNode, EETypeOptionalFieldsNode> _typeOptionalFields;
```

Comment:
True. I considered this but ended up not doing it. Your vote moves the scale towards taking `NodeFactory` out of the equation.